scipy/stats/tests/test_distributions.py

```diff
@@ -387,6 +387,11 @@ def test_ppf(self):
         expected = array([1.0, 2.0, 3.0])
         assert_array_almost_equal(vals, expected)
 
+    def test_ppf_underflow(self):
+        # neither of these should underflow
+        stats.geom.ppf(0.5, 1e-20)
```

Comment:
It would be better if the tests actually tested that the values that are returned are correct.  The second call, `stats.geom.ppf(1e-20, 0.5)`, doesn't trigger any problems in master, so calling it here doesn't actually verify that the function is working correctly.  The first call only raises a warning and returns `-inf` in master.